### DISCUSSION TITLE
Android bitwarden app can't connect to vaultwarden server

### DISCUSSION THREAD

**Discussion by sbhrepo** [dani-garcia/vaultwarden#3044] [Q&A]
for sometime I have docker vaultwarden hosted on synology with lets encrypt certificate. It worked fine no issues. I was able to connect from desktop, laptop and android phone.
few days ago I decided to upgrade the the vaultwarden image as new multiple version come between then to now.
I toke the vaultwarden down, backed up the data, removed the container and old image and download the latest image.
I setup the configuration and deployed the new container.
I was able to connect from my desktop and laptop with no issue and saw all my data. and the connection was secured (https connection all good).
but when I tried to connect from my android phone it was refused. it was try for around 30 second to a minute and tell me that it can't connect (the messages not detailed but got different messages in different attempts). I clean the cache and data from the app and try again, same problem. I uninstalled the app and reinstall it, same problem.
I toke 2 extra android devices, one the same mode and my and one different model. I installed bitwarden app on both android devices and both work with no problem.
only my device that was connected before the upgrade has the problem.
at this point I don't know how to continue and what I can do.
its not clear to me if it's a bad state on my device that prevent it working or some information on the database that was transferred to the upgraded/new server not like my device as some identity not much now.
I done many searches for this issue but I didn't find any solution to this problem.
I'll appreciate any suggestion\help available.

**Comment by BlackDex** [comment]
Have you tried to update to the latest version of Vaultwarden already?
The current version is v1.27.0. That solves all issues with the new clients recently released by Bitwarden.

**Reply by sbhrepo** [reply]
The all problem started when I upgraded vaultwarden to 1.26.0 from old version.
since you mention new version I just installed the new version 1.27.0
the problem persist all the clients working fine (including other android devices that I connected them after I encountered the problem with my android device), only the android phone that was connected to the old server can't connect.

**Reply by NoelGisler** [reply]
had the same problem on iOS

reinstalling the App fixed it for me, but i think on Android you can just clear the data in the Settings :)

**Reply by sbhrepo** [reply]
I did it many times with no success.
I delete app cache and app data - did not help.
then I uninstall the app, reboot the device and install it back - did not help.
at this point I think it some information that saved in the vaultwarden data folder hold some information about my device or something or some problem on the device regarding certificates. but regarding the certificates it not make sense, when I toke a new device that never connected to the old vaultwarden server before I didn't need to do anything to have secure connection. the only thing that make sense to me is some info in the data folder that not cleaned nicely and denied my phone to connect

**Comment by stefan0xC** [comment]
Are the date/time settings on your device correct? Is the Server URL set correctly?

Might be an issue with your device/OS. What respective versions are you using? Can you also provide a screenshot of the error message of the client?

You could also check with https://www.ssllabs.com/ssltest/ for potential issues/incompatibilities in your TLS setup.

**Reply by sbhrepo** [reply]
date/time is right and server URL set correctly.
it may be issue in the OS and in the end I may need to reset factory my phone (but I prefer to avoid it if possible)

server side version: 1.27.0
client version: latest from play store.

I can't take a screenshot as the app protected against it (security policy of the app). but the message that I get is: "An error has occurred. Exception message: The operation was canceled."
I can't use the website as it internal to my network only. but I know it's not issue as I can connect from 2 other devices and from multiple computers and laptop's

**Reply by stefan0xC** [reply]
Since the website is internal to your network only: do you access the network via WiFi or do you have some form of VPN? Do you have mobile data on the device enabled or disabled?

**Reply by sbhrepo** [reply]
the desktop access via wired LAN connection, the android devices and laptop via wifi when I'm at home and when I out, I can access it via VPN. it work for all devices beside my android device and not only via the bitwarden app, it fail to connect via the web browser as well.
I start to believe that something go wrong on the android OS system of my device.

**Reply by BlackDex** [reply]
You probably do not provide the full cert chain. Check the certs  as suggested already. See https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS#check-if-certificate-is-valid for some tips.

**Reply by sbhrepo** [reply]
I checked full cert chain like the show in the link that you provided. and its good. 
I did not expect it to be bad as it work with other 2 android devices without any problem. the problem is only with the device that was connected to the server before the upgrade of the server.

**Reply by BlackDex** [reply]
Try to clear cache and data. Uninstall the Bitwarden client. Reboot the phone, install the client again. In that exact order.
Also, you could try to install the f-droid version, which doesn't support the push-service, but that doesn't work for Vaultwarden anyway.
You can either download it via f-droid (or [Droid-ify](https://github.com/Iamlooker/Droid-ify/releases/tag/v0.5.1)) or download it directly from the repo from Bitwarden https://github.com/bitwarden/mobile/releases/tag/v2022.11.0

**Reply by sbhrepo** [reply]
I clean the cache then data uninstalled the app and reboot the device then install the app. same problem. I don't think it's app problem since I see the same problem if I open the browser on this device and try to connect from the browser. (all the other devices that the app working well, it working from both browser and app).

**Reply by BlackDex** [reply]
I guess that Android is broken then  either full hardware/factory reset, or don't use that Android device anymore.

**Reply by sbhrepo** [reply]
If I just know which certificates to delete from this device it may help. but I probably will factory reset it. it suck as I will need to setup a lot of staff on it after

**Reply by BlackDex** [reply]
Well, i think this place is not the best place to ask these questions. Better suited for some Android forum like xda-developers or something.

**Comment by skid9000** [comment]
I don't know how the docker image is made but your issue seems quite similar to this old one... https://vaultwarden.discourse.group/t/solved-android-client-can-t-sync/358/45

**Comment by shiyunjin** [comment]
I encountered the same situation，and I debug bitwarden mobile project in android got in `/sync` path, bitwarden mobile app decode json exception

```
Error converting value {null} to type 'Bit.Core.Enums.SecureNoteType'. Path 'Ciphers[9].SecureNote.Type', line 1, position 23626.
```

```
in /sync json:

            "SecureNote": {
                "Type": null
            },

another:

"SecureNote": null,
```

**Reply by delcroip** [reply]
what is the normal value ?

**Reply by BlackDex** [reply]
Not sure why you are commenting on a 2 year old discussion. This is totally outdated
